tests/test-derivation.nix

```diff
@@ -79,20 +78,60 @@ let
     # At least one of these should be undefined
     assert lib.assertMsg (tests == [ ] || module == null) "Both module & tests can't be supplied";
     let
+      helpers = import ../lib/helpers.nix {
+        inherit pkgs lib;
+        _nixvimTests = true;
+      };
       moduleAsList = lib.optional (module != null) { inherit module name; };
       moduleList = tests ++ moduleAsList;
       testDerivations = builtins.map (
         {
           module,
           name,
           dontRun ? false,
-        }:
-        {
-          derivation = makeNixvimWithModule {
-            inherit pkgs module extraSpecialArgs;
-            _nixvimTests = true;
+        }@args:
+        let
+          result = lib.evalModules {
+            modules = [
+              module
+              ./_module.nix
+              ../modules/top-level
+            ] ++ lib.optional (args ? dontRun) { tests.dontRun = lib.mkDefault dontRun; };
```

Comment:
It'd be simpler to drop the `dontTest` attr in moduleList. If we keep the function arg/attr then we need to pass it in to the module config here using `mkDefault`.

@traxys is dropping the attr/fn-arg a breaking change? I assume yes, but probably not an important one?

It'd be nice to have `dontRun` _only_ be a module option (other than in `mkTestDerivationFromNvim`)